lib/unicode.ml

```diff
@@ -126,26 +126,29 @@ let utf8_of_unicode n =
   if n < 128 then
     String.make 1 (Char.chr n)
   else if n < 2048 then
-    let s = String.make 2 (Char.chr (128 + n mod 64)) in
-    begin
-      s.[0] <- Char.chr (192 + n / 64);
-      s
-    end
+    String.init 2 (fun idx ->
+        match idx with
+        | 0 -> Char.chr (192 + n / 64)
+        | 1 -> Char.chr (128 + n mod 64)
+        | _ -> 'x'
+      )
   else if n < 65536 then
-    let s = String.make 3 (Char.chr (128 + n mod 64)) in
-    begin
-      s.[1] <- Char.chr (128 + (n / 64) mod 64);
-      s.[0] <- Char.chr (224 + n / 4096);
-      s
-    end
+    String.init 3 (fun idx ->
+        match idx with
+        | 0 -> Char.chr (224 + n / 4096)
+        | 1 -> Char.chr (128 + (n / 64) mod 64)
+        | 2 -> Char.chr (128 + n mod 64)
+        | _ -> 'x'
```

Comment:
I am no expert in Unicode and I don't have any preference, (well, I'd rather not have this code in Coq at all but depend on an external library)

I guess that anything that works with immutable strings should be fine for the purposes of this patch.